### ISSUE TITLE
SQL Authentication docs are contradictory and incomplete

### ISSUE THREAD

**Issue by Neutrino-Sunset** [MicrosoftDocs/sql-docs#9723]
This page [here](https://learn.microsoft.com/en-us/azure/azure-sql/database/security-best-practice?view=azuresql) repeatedly stresses that as a best practice Azure AD authentication should be preferred to SQL authentication.

But this other page [here](https://learn.microsoft.com/en-us/azure/azure-sql/database/authentication-azure-ad-logins?view=azuresql) highlights that Azure AD service principals are in public preview (and therefore not a recommended production configuration) for Azure SQL databases.

Your documentation should not recommend configurations that are not even supported. Some clarification around this is required.

In addition to that the `sqlcmd` version installed in Ubuntu devops pipelines is the ODBC version which also does not even support logging in via service principal anyway.

In short, you could read the entire Azure SQL documentation and still not have a clue how to actually configure a devops pipeline to create an SQL login for an application.

**Comment by nofield** [comment]
Hi @Neutrino-Sunset, thanks for your feedback!

Microsoft Entra (formerly Azure AD) authentication is absolutely preferred to SQL authentication!
To your first point- Microsoft Entra service principals are generally available for Azure SQL databases as users. That notice you mention is about the private preview of **server** principals support, or logins on a logical server, using Microsoft Entra identities (users, groups, and service principals).
I realize that the two terms are very similar. We can take an action item to clarify that language.

I can't speak to the Ubuntu devops pipelines specifically. I'll try to find the right contact for you with that.

**Comment by VanMSFT** [comment]
Closing as answered. #please-close
